gitpod-shared/src/features.ts

```diff
@@ -387,12 +387,36 @@ export async function registerWorkspaceTimeout(context: GitpodExtensionContext):
 			}
 		});
 		try {
-			const result = await context.gitpod.server.setWorkspaceTimeout(context.info.workspaceId, '180m');
-			if (result.resetTimeoutOnWorkspaces?.length > 0) {
-				vscode.window.showWarningMessage('Workspace timeout has been extended to three hours. This reset the workspace timeout for other workspaces.');
-			} else {
-				vscode.window.showInformationMessage('Workspace timeout has been extended to three hours.');
+			await context.gitpod.server.setWorkspaceTimeout(context.info.workspaceId, '180m');
+			vscode.window.showWarningMessage(`Workspace timeout has been extended to 180m.`);
+		} catch (err) {
+			vscode.window.showErrorMessage(`Cannot extend workspace timeout: ${err.toString()}`);
+		}
+	}));
+
+
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.setWorkspaceTimeout', async () => {
+		const timeout = await vscode.window.showInputBox({
+			placeHolder: 'Please input the timeout time, such as 30m, 1h, 2h, 3h',
+			validateInput: (value: string) => {
+				if (!WorkspaceTimeoutDuration.validate(value)) {
+					return 'Please input the timeout time, such as 30m, 1h, 2h, 3h';
+				}
+				return '';
+			}
+		});
+		if (!timeout) {
+			return;
+		}
+		context.fireAnalyticsEvent({
+			eventName: 'vscode_execute_command_gitpod_workspace',
+			properties: {
+				action: 'configure-timeout'
 			}
+		});
+		try {
+			await context.gitpod.server.setWorkspaceTimeout(context.info.workspaceId, timeout);
+			vscode.window.showWarningMessage(`Workspace timeout has been extended to ${timeout}.`);
```

Comment:
If we store the `SetWorkspaceTimeoutResult` above into a variable, we can use its `humanReadableDuration` to print a human-readable value. 

> **Note**: this requires updating `@gitpod/gitpod-protocol` to `0.1.5-main.6549` or newer.